lib/cache/entry.js

```diff
@@ -274,6 +274,10 @@ class CacheEntry {
       const cacheWritePromise = new Promise((resolve, reject) => {
         cacheWriteResolve = resolve
         cacheWriteReject = reject
+      }).catch((err) => {
```

Comment:
This would be from when we called (reject) so why not emit it there?

This may be a nitpick.

Reply:
line 302
```
cacheStream.promise().then(cacheWriteResolve, cacheWriteReject)
```

the error function should emit imho.

```
cacheStream.promise().then(cacheWriteResolve, err => {
         if (body) {
          body.emit('error', err)
        }
   cacheWriteReject(err)
})
```
(typed on the website, syntax may be all wrong).

Reply:
Also dropped the `if (body)` statement as suggested since it was not necessary.